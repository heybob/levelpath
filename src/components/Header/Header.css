.header {
    background: black;
    color: white;
    position: sticky;
    top: 0;
    padding: 10px 0;
}
h1, h2 {
    margin: 0;
}