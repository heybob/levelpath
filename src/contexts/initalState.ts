import { Text } from "../shared/constants"

export const initialState = {
    isLoading: false, 
    data: [[]], 
    paginatedData: [], 
    pageNum: 1, 
    totalPages: 0,
    pageSize: 50, 
    today: new Date('December 17, 1995 03:24:00'),
    apiError: false,
    pageTitle: Text.headerTitle
}