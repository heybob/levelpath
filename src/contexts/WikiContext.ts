import React from "react";
import { createContext } from "react";
import { initialState } from "./initalState";
import { WikiState } from "../shared/types";

export const  WikiContext = createContext<{ 
    state: WikiState; dispatch: React.Dispatch<any>;}>(
        {state: initialState, dispatch: () => null});