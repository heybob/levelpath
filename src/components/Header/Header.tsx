import './Header.css'
import { useContext } from 'react'
import { WikiContext } from '../../contexts/WikiContext'

export function Header() {
    const {state} = useContext(WikiContext);
    const {today, pageTitle} = state;

    return (
    <div className='header'>
        <h1>{pageTitle} </h1>
        <h2>{today.getMonth() + 1}/{today.getDate()}/{today.getFullYear()}</h2>
    </div>
    )
}