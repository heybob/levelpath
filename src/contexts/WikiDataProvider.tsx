import React from 'react'
import { useReducer} from "react";
import { wikiReducer } from "../components/App/reducer";
import { initialState } from "./initalState";
import { WikiContext } from "./WikiContext";

type Props = {
    children: React.ReactNode
  }

export function WikiDataProvider({children}: Props) {

    const [state, dispatch] = useReducer(wikiReducer, initialState);

    return (
        <WikiContext.Provider value={{state, dispatch}}>{children}</WikiContext.Provider>
    );

}