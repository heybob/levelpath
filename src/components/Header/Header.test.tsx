import renderer from 'react-test-renderer';
import { WikiDataProvider } from '../../contexts/WikiDataProvider';
import { Header } from "./Header"

describe('Testing Header Component', () => {
  it('It should render Header Title', () => {
      const tree = renderer.create(
      <WikiDataProvider>
            <Header />
        </WikiDataProvider>
      ).toJSON()
    expect(tree).toMatchSnapshot();
  });
});
