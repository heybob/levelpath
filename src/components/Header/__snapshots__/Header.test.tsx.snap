// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header Component It should render Header Title 1`] = `
<div
  className="header"
>
  <h1>
    Wikipedia Presents: Born On This Day
     
  </h1>
  <h2>
    12
    /
    17
    /
    1995
  </h2>
</div>
`;
